src/libs/setEnvironment/index.android.js

```diff
@@ -0,0 +1,41 @@
+import lodashGet from 'lodash/get';
+import Config from 'react-native-config';
+import Onyx from 'react-native-onyx';
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import {version} from '../../../package.json';
+
+export default function setEnvironment() {
+    if (lodashGet(Config, 'ENVIRONMENT', CONST.ENVIRONMENT.DEV) === CONST.ENVIRONMENT.DEV) {
+        Onyx.set(ONYXKEYS.ENVIRONMENT, CONST.ENVIRONMENT.DEV);
+        return;
+    }
+
+    // Since we promote staging builds to production without creating a new build, check the google play listing to
+    // see if the current version is greater than the production one
+    try {
+        fetch(CONST.PLAY_STORE_URL)
+            .then(res => res.text())
+            .then((text) => {
+                const match = text.match(/<span[^>]+class="htlgb"[^>]*>([-\d.]+)<\/span>/);
```

Comment:
This seems like it could easily break if Google decides to change the name of the class. Can't think of anything better though, I did a quick search and it doesn't seem like they have an API to query an app's version number.
Maybe it's overkill but could we store the version number somewhere (tbd where) when we make a new release, and then make a new API command on expensify.com to retrieve it?

Reply:
That was something I thought of doing as well since it is one of the only other options I found. I ultimately decided against it since that HTML hasn't been updated in >5 years, and even if it was updated it'd be easy to update the code. Also, having the version somewhere means we have to update expensify.com (or wherever we store that value) every time we deploy the app to production, which seemed like overkill since Google is already giving us that data on the play store page. If we think that's a better solution though, I can have a look at implementing that

Reply:
Cool, good to know it hasn't changed in over 5 years. And yes, it should be easy to update if they change the structure of the play store page.